allihoopa-sdk/src/main/java/com/allihoopa/sdk/AllihoopaPiece.java

```
@@ -236,4 +257,28 @@ public void getUrlWithFormat(String format, AudioWithFormatCallback callback) {
         void onComplete(@Nullable String url);
         void onError(Exception e);
     }
+
+    public static final class Attachment {
+        public URL attachmentURL;
+        public String attachmentMimeType;
+
+        public Attachment(URL attachmentURL, String attachmentMimeType) {
+            attachmentURL = attachmentURL;
+            attachmentMimeType = attachmentMimeType;
+        }
+
+        public static Attachment fromJSON(JSONObject attachmentObj, String attachmentMimeType) {
+            if (attachmentObj == null) {
+                return null;
+            }
+
+            try {
+                URL attachmentURL = new URL(attachmentObj.getString("url"));
+                AllihoopaPiece.Attachment attachment = new AllihoopaPiece.Attachment(attachmentURL, attachmentMimeType);
+                return attachment;
+            } catch (MalformedURLException | JSONException e) {
+                return null;
```

I'm not sure it's a good idea to swallow exceptions here - if I was to use this class I would really like to know if my JSON failed to deserialize.